src/layouts/instructions.py

```diff
@@ -0,0 +1,53 @@
+"""Layouts for dex instructions data."""
+from construct import Bytes, Int8ul, Int16ul, Int32ul, Int64ul  # type: ignore
+from construct import Struct as cStruct
+from construct import Switch
+
+from .slab import KEY
+
+INITIALIZE_MARKET = cStruct(
+    "baseLotSize" / Int64ul,
+    "quoteLotSize" / Int64ul,
+    "feeRateBps" / Int16ul,
+    "vaultSignerNonce" / Int64ul,
+    "quoteDustThreshold" / Int64ul,
+)
+
+NEW_ORDER = cStruct(
+    "side" / Int32ul,  # Enum
+    "limitPrice" / Int64ul,
+    "maxQuantity" / Int64ul,
+    "orderType" / Int32ul,  # Enum
+    "clientId" / Int64ul,
+)
+
+MATCH_ORDERS = cStruct("limit" / Int16ul)
+
+CONSUME_EVENTS = cStruct("limit" / Int16ul)
+
+CANCEL_ORDER = cStruct(
+    "side" / Int32ul,  # Enum
+    "orderId" / KEY,
+    "openOrders" / Bytes(32),
+    "openOrdersSlot" / Int8ul,
+)
+
+CANCEL_ORDER_BY_CLIENTID = cStruct("cancelOrderByClientId" / Int64ul)
+
+INSTRUCTIONS_LAYOUT = cStruct(
+    "version" / Int8ul,
+    "instruction" / Int8ul,
```

Comment:
This indicates the type of the instruction, so maybe it's more appropriate to be called `instruction_type` ?